### PR TITLE
Clean up web extension action menu popup 

### PR DESCRIPTION
**Pull request by psymoon**
- Remove onActionPopupToggleOverride and add openPopupInTab flag.
- Add store flow to observe for `extension.popUpSession` change when it's not null in sample browser.
- Fix web extension menu item label view being clickable; only the container should be.

This cleans up and makes it easier for Fenix integration as well as easier to deprecate `WebExtensionSupport` class later down the road when browser store refactor is complete.

### CODE REVIEW

File: samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt [hunk 1]
```diff
@@ -43,6 +53,32 @@ open class BrowserActivity : AppCompatActivity(), ComponentCallbacks2 {
         }
     }
 
+    @kotlinx.coroutines.ExperimentalCoroutinesApi
+    override fun onStart() {
+        super.onStart()
+        scope = components.store.flowScoped { flow ->
+            flow.ifChanged { it.extensions }
+                .map { it.extensions.filterValues { extension -> extension.popupSession != null } }
+                .collect { pair ->
```
Comment (psymoon) [inline samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt]: TODO rename; this is not a pair...

File: components/browser/menu/src/test/java/mozilla/components/browser/menu/WebExtensionBrowserMenuTest.kt [hunk 2]
```diff
@@ -59,6 +59,7 @@ class WebExtensionBrowserMenuTest {
             "browser_action" to WebExtensionState(
                 "browser_action",
                 "url",
+                "",
```
Comment (csadilek) [inline components/browser/menu/src/test/java/mozilla/components/browser/menu/WebExtensionBrowserMenuTest.kt]: nit: Might as well put a "title" in there :)

File: samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt (hunk 1 above)
Comment (csadilek) [reply samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt]: Yeah, maybe just extensionState?

Comment (csadilek) [reply samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt]: nit: I would (also for Fenix) move this to a private function to make it clear what this does

```
webExtScope = observeWebExtensionPopups()
```

File: samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt [hunk 3]
```diff
@@ -43,6 +53,32 @@ open class BrowserActivity : AppCompatActivity(), ComponentCallbacks2 {
...
+                    if (pair.values.isNotEmpty()) {
+                        // There could only be one active popupSession instance at a time
```
Comment (csadilek) [inline samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt]: nit: // We currently limit to one active popup session at a time ?

File: samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt (hunk 1 above)
Comment (psymoon) [reply samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt]: I like it

File: samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt [hunk 4]
```diff
@@ -43,6 +53,16 @@ open class BrowserActivity : AppCompatActivity(), ComponentCallbacks2 {
         }
     }
 
+    override fun onStart() {
+        super.onStart()
+        scope = observeWebExtensionPopups()
```
Comment (csadilek) [inline samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt]: nit: would give the scope a name "webExtScope" or similar :)

File: samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt [hunk 5]
```diff
@@ -70,4 +90,23 @@ open class BrowserActivity : AppCompatActivity(), ComponentCallbacks2 {
         components.sessionManager.onLowMemory()
         components.icons.onLowMemory()
     }
+
+    private fun observeWebExtensionPopups(): CoroutineScope {
+        return components.store.flowScoped { flow ->
+            flow.ifChanged { it.extensions }
+                .map { it.extensions.filterValues { extension -> extension.popupSession != null } }
```
Comment (csadilek) [inline samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt]: I think here, after the map, we need another empty `.ifChanged()` to make sure that an unrelated change to the global extension state doesn't get collected, otherwise we'd try to display the same popup again.

In other words, we only want to collect if we found a new non-null popupSession.

**Comment by psymoon** [comment]
bors r=csadilek

**Comment by psymoon** [comment]
bors retry
